crates/svm-runtime/src/import.rs

```diff
@@ -55,13 +56,20 @@ impl ExternImport {
             let returns_types = self.returns.clone();
             let func = self.func;
 
+            #[derive(wasmer::WasmerEnv)]
+            struct WrapperEnv {
+                func_env: *mut svm_env_t,
+            }
+
             let wrapper_callback =
-                move |env: &mut *mut svm_env_t, args: &[Val]| -> Result<Vec<Val>, RuntimeError> {
+                move |env: &WrapperEnv, args: &[Val]| -> Result<Vec<Val>, RuntimeError> {
                     let args: Vec<WasmValue> = wasmer_vals_to_wasm_vals(args)?;
                     let args: svm_byte_array = (WASMER_ARGS_STR, args).into();
 
                     let mut results = svm_ffi::alloc_wasm_values(returns_types.len());
-                    let err: *mut svm_byte_array = func(*env, &args, &mut results);
+
+                    let env = env.func_env as *mut svm_env_t;
```

Comment:
Is this casting necessary?

Reply:
yes, `func` expects the first argument to be of `*mut svm_byte_array`
(see the type `svm_func_callback_t`_